### PR TITLE
Add missing methods to TypeScript definitions

### PR DESCRIPTION
**Pull request by Shaptic**
Self-explanatory, see diff.

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -1058,7 +1059,9 @@ export class TransactionBuilder {
     options?: TransactionBuilder.TransactionBuilderOptions
   );
   addOperation(operation: xdr.Operation): this;
+  addOperationAt(op: xdr.Operation, i: number): this;
```
Comment (sreuland) [inline types/index.d.ts:1062]: is it getting to point where it's worthwhile to add `ts` tests, `index.d.test.ts`? how do we know `addOperation` will work when used by a client, or is there some base validation done in build pipeline?

Comment (Shaptic) [reply types/index.d.ts:1062]: There's JS unit tests for it but you're right that we should generally add more to the `types/test.ts` file. Unfortunately said tests only fail when you, well, add to them, and if you're adding to them then you're adding to the `index.d.ts` file anyway, so it wouldn't have prevented this slip-up (which came from an external contributor anyway).

Comment (Shaptic) [reply types/index.d.ts:1062]: Added the typing tests in c5c5f9b!

**Review by sreuland — APPROVED** [review]
lgtm, and thanks for insight there on ts testing mechanics.

**Pull request merged**